diff --git a/src/authunix_prot.c b/src/authunix_prot.c
index bf76a9d..62bdf37 100644
--- a/src/authunix_prot.c
+++ b/src/authunix_prot.c
@@ -56,10 +56,17 @@ xdr_authunix_parms(xdrs, p)
 
 	if (xdr_u_long(xdrs, &(p->aup_time))
 	    && xdr_string(xdrs, &(p->aup_machname), MAX_MACHINE_NAME)
+#ifdef __CYGWIN__
+	    && xdr_u_long(xdrs, &(p->aup_uid))
+	    && xdr_u_long(xdrs, &(p->aup_gid))
+	    && xdr_array(xdrs, (caddr_t *)&(p->aup_gids),
+		    &(p->aup_len), NGRPS, sizeof(unsigned long), (xdrproc_t)xdr_u_long) ) {
+#else
 	    && xdr_u_int(xdrs, &(p->aup_uid))
 	    && xdr_u_int(xdrs, &(p->aup_gid))
 	    && xdr_array(xdrs, (caddr_t *)&(p->aup_gids),
 		    &(p->aup_len), NGRPS, sizeof(int), (xdrproc_t)xdr_int) ) {
+#endif
 		return (TRUE);
 	}
 	return (FALSE);
diff --git a/src/rpcb_clnt.c b/src/rpcb_clnt.c
index 4a3e96c..41b33f8 100644
--- a/src/rpcb_clnt.c
+++ b/src/rpcb_clnt.c
@@ -557,7 +557,11 @@ rpcb_set(program, version, nconf, address)
 	 * completeness.  For non-unix platforms, perhaps some other
 	 * string or an empty string can be sent.
 	 */
+#ifdef __CYGWIN__
+	(void) snprintf(uidbuf, sizeof uidbuf, "%lu", geteuid());
+#else
 	(void) snprintf(uidbuf, sizeof uidbuf, "%d", geteuid());
+#endif
 	parms.r_owner = uidbuf;
 
 	CLNT_CALL(client, (rpcproc_t)RPCBPROC_SET, (xdrproc_t) xdr_rpcb,
@@ -601,7 +605,11 @@ rpcb_unset(program, version, nconf)
 	}
 	/*LINTED const castaway*/
 	parms.r_addr = (char *) &nullstring[0];
+#ifdef __CYGWIN__
+	(void) snprintf(uidbuf, sizeof uidbuf, "%lu", geteuid());
+#else
 	(void) snprintf(uidbuf, sizeof uidbuf, "%d", geteuid());
+#endif
 	parms.r_owner = uidbuf;
 
 	CLNT_CALL(client, (rpcproc_t)RPCBPROC_UNSET, (xdrproc_t) xdr_rpcb,
