--- origsrc/libtirpc-0.2.1/src/getrpcent.c	2009-11-30 09:04:55.000000000 -0500
+++ src/libtirpc-0.2.1/src/getrpcent.c	2010-08-18 22:12:57.844600000 -0400
@@ -87,7 +87,6 @@ _rpcdata()
 	return (d);
 }
 
-#ifdef GQ
 struct rpcent *
 getrpcbynumber(number)
 	int number;
@@ -136,7 +135,7 @@ no_yp:
 
 struct rpcent *
 getrpcbyname(name)
-	char *name;
+	const char *name;
 {
 	struct rpcent *rpc = NULL;
 	char **rp;
@@ -156,7 +155,6 @@ done:
 	endrpcent();
 	return (rpc);
 }
-#endif /* GQ */
 
 void
 setrpcent(f)
